tests/library/test_utils.py

```diff
@@ -1,130 +1,88 @@
 
 
 class TestUtils(unittest.TestCase):
-    def test_get_minutes_english_description(self):
-        text = "1 hour 15 mins"
-        self.assertEqual(75, get_minutes(text))
-
-    def test_get_minutes_english_description_with_and(self):
-        text = "1h and 15mins"
-        self.assertEqual(75, get_minutes(text))
-
-    def test_get_minutes_english_abbreviation(self):
-        text = "3h10m"
-        self.assertEqual(190, get_minutes(text))
-
-    def test_get_minutes_short_iso_format(self):
-        text = "PT2H30M"
-        self.assertEqual(150, get_minutes(text))
-
-    def test_get_minutes_long_iso_format(self):
-        text = "P0DT1H10M"
-        self.assertEqual(70, get_minutes(text))
-
-    def test_get_minutes_int_in_string_literal(self):
-        text = "90"
-        self.assertEqual(90, get_minutes(text))
-
-    def test_get_minutes_fraction_in_hours_with_dot_notation(self):
-        text = "1.5 hours"
-        self.assertEqual(90, get_minutes(text))
-
-    def test_get_minutes_integer_days(self):
-        text = "2 days"
-        self.assertEqual(2880, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_unicode_character_halves(self):
-        text = "1½ hours"
-        self.assertEqual(90, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_unicode_character_three_fours(self):
-        text = "1¾ hours"
-        self.assertEqual(105, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_unicode_character_one_fours(self):
-        text = "1¼ hours"
-        self.assertEqual(75, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_unicode_character_two_thirds(self):
-        text = "1⅔ hours"
-        self.assertEqual(100, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_digits_with_slash(self):
-        text = "1 1/2 hours"
-        self.assertEqual(90, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_digits_with_slash_three_fours(self):
-        text = "1 3/4 hours"
-        self.assertEqual(105, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_digits_with_slash_one_fours(self):
-        text = "1 1/4 hours"
-        self.assertEqual(75, get_minutes(text))
-
-    def test_get_minutes_fraction_with_fraction_digits_with_slash_two_thirds(self):
-        text = "1 2/3 hours"
-        self.assertEqual(100, get_minutes(text))
-
-    def test_get_minutes_handles_dashes(self):
-        text = "15 - 20 minutes"
-        self.assertEqual(20, get_minutes(text))
-
-    def test_get_minutes_handles_to(self):
-        text = "15 to 20 minutes"
-        self.assertEqual(20, get_minutes(text))
-
-    def test_get_minutes_imprecise_description(self):
-        text = "Pá-Pum"
-        self.assertEqual(None, get_minutes(text))
+    @classmethod
+    def setUpClass(cls):
+        cls.iso8601_fixtures = {
+            "PT1H": 60,
+            "PT20M": 20,
+            "PT2H10M": 130,
+            "PT0H9M30S": 10,
```

Comment:
Let me know if you'd prefer me to handle this in a separate pull request, but: during code review of #1037 I've noticed that we don't treat empty ISO durations as `None` like we do for other zero/empty times.

Could we add a test case and fixup to do that?  (doing so breaks three existing test expectations for `0` minutes, but I think that's a sign that those should be changed to expect `None`)

Reply:
Ah, yep.  Could that be due to the removal of the `isodate.parse_duration` call?  I think we should keep that in place, but return `None` if the time duration it returns is zero.

Reply:
That was it! Fixed